### PR TITLE
Extend documentation for plotting functionality

### LINKED ISSUE #802: DOC: extend documentation for plotting functionality provided through MetricFrame
#### Describe the issue linked to the documentation

@hildeweerts [pointed out](https://github.com/fairlearn/fairlearn/pull/770#pullrequestreview-657627887) that our new documentation around plotting using `fairlearn.metrics.MetricFrame` could elaborate a little more on what you can do with the `matplotlib` plotting options.

#### Suggest a potential alternative/fix

The assessment user guide section on plotting should discuss at least the following options:
- `ylim` for bar charts
- how to adjust the colors
- how to use various kinds of charts, e.g., pie charts for the `count` metric, and bar charts for `accuracy_score`
- how to plot multiple metrics (is already included, but could be made clearer)

### PR DESCRIPTION
**Pull request by bthng**
Had a go at Issue #802, would like to get some feedback if this is the right way forward before I attempt to add more. Thank you!

Resolves #802

### CODE REVIEW

File: docs/user_guide/assessment.rst
Comment (romanlutz) [inline docs/user_guide/assessment.rst:429]: This is excellent! However, when using `code-block` the code isn't actually run during the doc build. Instead, I suggest using the same technique I used for the "other" plotting doc section (in this same file towards the end) using `literalinclude` from the `plot_quickstart.py` file. I would suggest extending it a bit and adding these other plot representations (like pie), but that would also mean that you'd have to restrict the existing occurrences of `literalinclude` for that file, but I can point you there when we get to that point.

File: docs/user_guide/assessment.rst
Comment (hildeweerts) [inline docs/user_guide/assessment.rst:414]: A few small suggestions:
* yaxis -> y-axis
* ... which sometimes can be misleading, hence it is ... -> which can be misleading. Therefore it is

File: docs/user_guide/assessment.rst
Comment (hildeweerts) [reply docs/user_guide/assessment.rst:429]: Question: do we have specific cases in mind in which a pie chart is preferable over a bar chart? In the dataviz community pie charts are usually scrutinized quite a bit because they can be tricky to read (see e.g., [this blogpost](https://www.data-to-viz.com/caveat/pie.html): TDLR, people are bad at reading angles).

File: docs/user_guide/assessment.rst
Comment (bthng) [reply docs/user_guide/assessment.rst]: ah I think I created this before the latest merge, let me rebase and work off from there, thank you!

File: docs/user_guide/assessment.rst
Comment (bthng) [reply docs/user_guide/assessment.rst]: Got it! Adjusted the documentation to use `literal include` and also amended the comment around the `kind` param (pie/bar chart) to be more generic - simply showing the reader of the option to switch up the various chart types

File: docs/user_guide/assessment.rst
Comment (romanlutz) [inline docs/user_guide/assessment.rst:419]: Does this way of underlining work? I think `~` works for sure. Our doc build preview doesn't currently work so I haven't checked yet, but please briefly check and let us know if this way works, too.

File: docs/user_guide/assessment.rst
Comment (romanlutz) [inline docs/user_guide/assessment.rst:421]: nit: can you use `:code:` before `ylim`? That way it should render as code rather than in italics. [Hopefully I'm not mixing things up, but feel free to tell me I'm wrong!]

File: docs/user_guide/assessment.rst
Comment (romanlutz) [inline docs/user_guide/assessment.rst:410]: @riedgar-ms was rather attached to the `:start-after:` tag. Perhaps we can keep that and use `:end-before` for the end? Or is this acceptable to you @riedgar-ms ?
I'm personally fine with  `:lines:`

File: docs/user_guide/assessment.rst
Comment (romanlutz) [reply docs/user_guide/assessment.rst:421]: same for  other code occurrences below, of course 😄

File: docs/user_guide/assessment.rst
Comment (romanlutz) [inline docs/user_guide/assessment.rst:456]: I think you can link to this directly with 
```
:meth:`pandas.DataFrame.plot`
```
but this is really nitpicky and not a huge improvement. If you want to ignore it just resolve 😄

**Review by romanlutz — COMMENTED** [review]
Thanks @bthng for the adjustments and putting all the code into the examples file! This is exactly what I was looking for. The remaining are just small things. I'm happy to approve/merge after that 😄 Thanks again for your awesome contributions!!!

File: docs/user_guide/assessment.rst
Comment (bthng) [reply docs/user_guide/assessment.rst:419]: I still couldn't get my local to build with the theme but I referred to the [sphinx docs](https://documentation-style-guide-sphinx.readthedocs.io/en/latest/style-guide.html#headings). Here is my local build without the theme, I meant for the customisation headings to be H6, under the `Plotting Grouped Metrics` section (H5)

image
image

File: docs/user_guide/assessment.rst
Comment (bthng) [reply docs/user_guide/assessment.rst:421]: cool! didn't notice the diff between italics and code, changed!

File: docs/user_guide/assessment.rst
Comment (bthng) [reply docs/user_guide/assessment.rst]: oh that's cool! learning sth new everyday, changed!

File: docs/user_guide/assessment.rst
Comment (bthng) [reply docs/user_guide/assessment.rst:410]: sure not a problem, I switched them back!

File: docs/user_guide/assessment.rst
Comment (romanlutz) [reply docs/user_guide/assessment.rst]: It should be fixed now. There was a doc build issue that's now resolved as of today #815

Comment (romanlutz) [reply docs/user_guide/assessment.rst]: To elaborate: you may have to pull in the latest `main` branch. It might automatically just merge it, so I've kicked off the doc build again, but for yourself locally you'll need to pull it in.

**Comment by romanlutz** [comment]
I merged in the latest changes from main, particularly #815 since that is the fix for the issues you're seeing the doc build. I'm happy with the changes as far as I can tell, but will verify once the doc build finishes

File: docs/user_guide/assessment.rst
Comment (romanlutz) [reply docs/user_guide/assessment.rst]: Can you make sure the `'''''''` render like `~~~~~~~` when using in a header? Or better, just replace it for uniformity with other headers?

File: docs/user_guide/assessment.rst
Comment (romanlutz) [inline docs/user_guide/assessment.rst:433]: Sorry, I actually meant to add `:code:` only in the text, not in headers. It seems to be causing some issues when I run it locally.

Comment (romanlutz) [reply docs/user_guide/assessment.rst:433]: OK I'll take it back, perhaps it's the length of the header underline. But it's definitely struggling with the backquotes...

```
fairlearn\docs\user_guide\assessment.rst:434: WARNING: Title underline too short.

Customize Plots: :code:`colormap`
''''''''''''''''''''''''''''''''
```
I suggest replacing them with `~`

Comment (bthng) [reply docs/user_guide/assessment.rst:437]: sorry for the delay, got a bit held up last week. I reflected the feedback and it seems to be passing the build now. Does it look okay? Thank you!

**Review by romanlutz — APPROVED** [review]
This is ready to merge. I built it locally and exchanged the header underline to make it pass. The quickstart needed to get an `:end-before:` tag to avoid showing the customized code there (in addition to the user guide where we already explain it). @hildeweerts please merge unless you have further comments. This is what it looks like now:

image

image

image

image

File: docs/user_guide/assessment.rst
Comment (hildeweerts) [inline docs/user_guide/assessment.rst:459]: I think this should be "here is an example of a *pie* chart"?
```suggestion
argument. Here is an example of a pie chart.
```

File: docs/user_guide/assessment.rst
Comment (hildeweerts) [inline docs/user_guide/assessment.rst:427]: Nitpick: as the following paragraph starts with a header, can we replace the colon : with a dot . ? 
```suggestion
It is possible to customize the plots. Here are some common examples.
```

**Review by hildeweerts — CHANGES_REQUESTED** [review]
Thank you for adding the screen grabs @romanlutz.

This looks great, @bthng! I still have two tiny comments.

**Comment by hildeweerts** [comment]
Thanks again @bthng!

**Pull request merged**
